import-service/importfileparser/importfileparser.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+const { S3 } = require("aws-sdk");
+const { BUCKET, REGION } = require("../constants");
+const csv = require("csv-parser");
+
+module.exports.importFileParser = (event) => {
+  try {
+    const s3 = new S3({ region: REGION });
+    const { Records } = event;
+
+    for (const record of Records) {
+      const Key = record.s3.object.key;
+      const params = {
+        Bucket: BUCKET,
+        Key,
+      };
+
+      const s3Stream = s3.getObject(params).createReadStream();
+
+      s3Stream
+        .pipe(csv())
+        .on("data", (data) => {
+          console.log(JSON.stringify(data, null, 4));
+        })
+        .on("end", async () => {
```

Comment:
Async callbacks per se are OK as event handlers. But your lambda handler will not wait for them to finish.

Reply:
Hi! What do you mean that lambda will not wait for them to finish? But judging by the fact that these handlers are triggered, the lambda is waiting for them.

Reply:
oooh, i faced issue that you are talking about, it happend because of the main lambda function was async, i made it sync(by removing word async)) and it worked